### ISSUE TITLE
False positive on type alias with `Annotated` and multiple type arguments upon import

### ISSUE THREAD

**Issue by luwqz1** [microsoft/pyright#9273]
I made a type alias with `Annotated`, which has two arguments in the generic. Everything works correctly, but as soon as i import this type alias into another file, some strange behavior occurs. Unfortunately, I couldn't reproduce other examples, so I can't pinpoint the exact issue. Eric Traut, i hope you'll be able to figure it out, though it's possible that Pyright is working as intended. I'm always open to discussion!

File `foo.py`
```python
import typing

First = typing.TypeVar("First")
Meta = typing.TypeVar("Meta")

type FooType[First, Meta] = typing.Annotated[First, Meta]
Foo: typing.TypeAlias = FooType[First, Meta]

def bar(foo: Foo[str, int]): ...  # Ok
```

File `bar.py`
```python
from foo import Foo

def bar(foo: Foo[str, int]): ... # Error: Too many type arguments provided for "type[First@Foo]"; expected 1 but received 2
```

**Comment by erictraut** [comment]
I agree this is a bug. It will be addressed in the next release.

What you're doing here is pretty suspect though. Type parameters are intended to parameterize types. Here you're using a type parameter that appears only within an `Annotated` metadata expression, not within the base type expression. Any type argument supplied for the second type parameter will therefore be ignored by static type checkers. I guess this construct might be valid if you intend to use it with a runtime type checker or perform introspection of annotations in your own code.

**Comment by luwqz1** [comment]
> I guess this construct might be valid if you intend to use it with a runtime type checker or perform introspection of annotations in your own code.

Thank you very much for such a quick answer! You are absolutely right, i plan to use the second type parameter at runtime for other libraries that may perform checks/validation.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
